### ISSUE TITLE
Comparing a compiledRelease with the individual releases it was compiled from

### ISSUE THREAD

**Issue by duncandewhurst** [open-contracting/kingfisher#272]
@jpmckinney @robredpath and I briefly discussed whether it was necessary to support easier analysis across compiledReleases created by a transformation in Kingfisher and the individual releases which they were compiled from by having them under the same collection in the database, or whether it would be OK to have these loaded into separate collections in the database.

I reviewed the OCDS data queries, reports and visualization document and I don't think any of the queries expressed there would require this functionality, however I think this could be a useful feature for other users of the data where the original publisher only publishes releases.

For example, answering the question "did the contract value change after the original award?" would be easier to answer by comparing the compiledRelease with a 'contract' release, than it would be to answer from the individual releases alone (where updates to the contract value could occur under multiple release tags and where subsequent releases may choose to omit the contract value).

The other use I can think of for distinguishing between kingfisher-compiled releases and publisher-compiled releases (as opposed to just loading compiled releases from outside the DB and not knowing whether they were compiled by Kingfisher or the publisher) is to answer questions of the type "how many publishers are using the correct tag for compiled releases" etc.

**Comment by jpmckinney** [comment]
See additional notes from call at https://docs.google.com/document/d/1zeBhhSV9eLPoRu8cIy-0y4HNT02fCXNr3b6LdtyijVM/edit#heading=h.58pa2q9ekpxc

For the question "did the contract value change after the original award?", I think it would be best to create a versioned release (essentially the same steps as creating a compiled release), and then look at the relevant field, to see all its values at once. Using a compiled release in combination with individual releases seems like a more difficult way to answer the question.

For corpus-wide queries (like "how many publishers are using the correct tag for compiled releases"), I think each collection should indicate its source, as is already the case. If we, e.g., take the packages downloaded to disk by kingfisher-scrape, create compile releases, and stream them into the database with kingfisher-process, we would/should set the source for that new collection to some other value (e.g. `uk_contracts_finder-compiled`), so that we can identify it as not being directly from the source, when trying to answer the above question.

For the reasons I gave in the additional notes linked above, I think it's better to add new collections for any relevant alterations we make to the original source, rather than add more complexity to the concept of a collection – which might eventually need to track: which compiled releases are ours versus theirs; which releases have been upgraded to 1.1 (so that we don't need to write queries for both 1.0 and 1.1, but can still interrogate the 1.0 data); which individual releases have had, for example, strings coerced to numbers (so that we can reliably calculate indicators, generate histogram, etc. without all those tools having to implement coercion logic).

In short, I figure letting the analyst manage their collections with careful naming (or other metadata) is a better route than trying to make the collection smart enough to handle all the future needs.

**Comment by duncandewhurst** [comment]
My issue with this approach is that, from a helpdesk perspective, I think that conversion to 1.1 and compilation should be done by default for all sources loaded into kingfisher (albeit with an option to skip these transformations to support other use cases which might not need these features), so that when querying across multiple sources, which may have been loaded by other helpdesk analysts, you know what to expect.

Many types of analysis would be best performed against compiled releases (most coverage queries, many conformance and coherence queries) and since the majority of publishers only currently publish individual releases, it would be preferable to have a table in the database which contains kingfisher-compiled, 1.1 transformed releases for all sources.

> In short, I figure letting the analyst manage their collections with careful naming (or other metadata) is a better route than trying to make the collection smart enough to handle all the future needs.

I'm wary of introducing manual steps (collection naming or populating metadata) to the process of loading data at the risk of analysts being unsure of what they are looking at when using data loaded by another analyst.

@odscjames this is what we discussed yesterday (cc @kindly)

**Comment by jpmckinney** [comment]
I think we have the same goals; we're just in the process of agreeing how to get there.

For the helpdesk use case, I imagine the deployment of these tools will look like:

1. kingfisher-scrape and kingfisher-process are installed on the server
2. A configuration file (or environment variables) on the server indicates the storage method(s) for kingfisher-scrape to use. This will set the URL for the API endpoint of kingfisher-process to which data will be sent for processing, and will presumably also set a path to a directory in which to store the data on disk (in case we want to use the data with other tools, or simply to back up the original data).
3. Another configuration file (or environment variables) on the server indicates the pipeline(s) for kingfisher-process to run when it receives data via API. In this case, we might have one pipeline only: upgrade to 1.1, create compiled releases, and store both upgraded and compiled releases in one collection.

Those three steps are part of a one-time setup, which only needs to be figured out once, and which helpdesk analysts don't necessarily need to know anything about or ever touch, as the setup can be handled by developers.

Now, for a helpdesk analyst, their interaction with this system might be e.g.:

* logging into the server and running `scrapy runspider uk_contracts_finder`
* running `ssh user@server scrapy runspider uk_contracts_finder` from their local machine
* clicking a button in a web UI to run a scraper

Then, processed data will start to appear in the database, which the helpdesk analyst can query from notebooks.

----

I think this is as simple as possible – and you'll note that there are no unnecessary manual steps for the helpdesk analyst. You'll also note that the above doesn't build-in any specialized logic or defaults.
